File: CarWatchBelgium.Web/src/types/listing.ts
export interface MatchedListingDto {
  listingId: string
  savedSearchId: string
  userListingStateId?: string
  source: string
  externalId?: string
  url: string
  imageUrl?: string
  title: string
  make?: string
  model?: string
  price?: number
  currency: string
  previousPrice?: number
  priceChanged: boolean
  priceDropped: boolean
  priceDropAmount?: number
  mileageKm?: number
  year?: number
  fuelType: string
  transmission: string
  powerHp?: number
  sellerType: string
  city?: string
  countryCode: string
  isAvailable: boolean
  isSeen: boolean
  isFavorite: boolean
  isIgnored: boolean
  isNew: boolean
  firstSeenAtUtc: string
  lastSeenAtUtc: string
  lastPriceChangeAtUtc?: string
  matchedAtUtc: string
}

export interface ListingDetailsDto {
  listingId: string
  source: string
  externalId?: string
  url: string
  imageUrl?: string
  title: string
  make?: string
  model?: string
  price?: number
  currency: string
  mileageKm?: number
  year?: number
  fuelType: string
  transmission: string
  powerHp?: number
  sellerType: string
  city?: string
  countryCode: string
  isAvailable: boolean
  isSeen: boolean
  isFavorite: boolean
  isIgnored: boolean
  firstSeenAtUtc: string
  lastSeenAtUtc: string
  lastPriceChangeAtUtc?: string
  priceHistory: PriceHistoryDto[]
}

export interface PriceHistoryDto {
  id: string
  price: number
  currency: string
  capturedAtUtc: string
}

export interface UpdateListingStateRequest {
  isSeen?: boolean
  isFavorite?: boolean
  isIgnored?: boolean
}

File: CarWatchBelgium.Domain/Entities/Listing.cs
namespace CarWatchBelgium.Domain.Entities;

using CarWatchBelgium.Domain.Enums;

public class Listing
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public ListingSource Source { get; set; }
    public string? ExternalId { get; set; }
    public string FallbackHash { get; set; } = null!;
    public string Url { get; set; } = null!;
    public string? ImageUrl { get; set; }
    public string Title { get; set; } = null!;
    public string? Make { get; set; }
    public string? Model { get; set; }
    public int? Price { get; set; }
    public string Currency { get; set; } = "EUR";
    public int? MileageKm { get; set; }
    public int? Year { get; set; }
    public FuelType FuelType { get; set; } = FuelType.Unknown;
    public TransmissionType Transmission { get; set; } = TransmissionType.Unknown;
    public int? PowerHp { get; set; }
    public SellerType SellerType { get; set; } = SellerType.Unknown;
    public string? City { get; set; }
    public string CountryCode { get; set; } = "BE";
    public bool IsAvailable { get; set; } = true;
    public DateTime FirstSeenAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime LastSeenAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? LastPriceChangeAtUtc { get; set; }

    // Navigation properties
    public ICollection<PriceHistory> PriceHistory { get; set; } = new List<PriceHistory>();
    public ICollection<UserListingState> UserStates { get; set; } = new List<UserListingState>();
    public ICollection<SavedSearchMatch> SavedSearchMatches { get; set; } = new List<SavedSearchMatch>();
}
